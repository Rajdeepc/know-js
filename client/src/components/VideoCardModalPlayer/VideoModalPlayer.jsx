import React from "react";
import { Card, Row,Col,ButtonGroup,Modal, Button } from "react-bootstrap";
import moment from "moment";
import './videoplayer.scss'
import { RiYoutubeLine } from "react-icons/ri";
import { BsBookmarks } from "react-icons/bs";
import { FiSave } from "react-icons/fi";
const VideoModalPlayer = ({item}) => {
  return (
    <div className="video-modal-player">
      <iframe
        width="100%"
        height="400"
        allowfullscreen
        src={`https://www.youtube.com/embed/${item.id.videoId}`}
      ></iframe>
      <div className="video-info">
        <div className="video-title">
          <Row>
            <Col sm="10">
              <h4>{item.snippet.title}</h4>
            </Col>
            <Col sm="2" className="text-right">
              <ButtonGroup aria-label="Basic example">
                <Button variant="light">
                  <BsBookmarks size={20} />
                </Button>
                <Button variant="light">
                  <FiSave size={20} />
                </Button>
              </ButtonGroup>
            </Col>
          </Row>

          <Row>
            <Col>
              <p>{item.snippet.description} </p>
            </Col>
          </Row>
          <Row>
            <Col>
              <div className="channel">
              Channel: <a
                  target="_blank"
                  href={`https://www.youtube.com/channel/${item.snippet.channelId}`}
                >
                  {item.snippet.channelTitle}
                </a>
                </div>
                </Col>
                <Col>
                    Published At: {moment(item.snippet.publishedAt).format("DD/MM/YYYY")}
                </Col>
          </Row>
        </div>
      </div>
    </div>
  );
};

export default VideoModalPlayer;
