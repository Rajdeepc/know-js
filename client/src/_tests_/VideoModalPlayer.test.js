import React from "react";
import { render, cleanup } from "@testing-library/react";
import VideoModalPlayer from "../components/VideoCardModalPlayer/VideoModalPlayer";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

const props = {
    item :{
      id: {
        videoId: ''
      },
      snippet: {
        title: '',
        description:'',
        channelId:"",
        channelTitle:'',
        publishedAt:''
      }
    }
  
};

describe("<VideoModalPlayer component>", () => {
  test("should be defined", () => {
    expect(VideoModalPlayer).toBeDefined();
  });
  test("renders VideoModalPlayer Component", () => {
    const { container, rerender } = render(<VideoModalPlayer {...props} />);
    rerender(<VideoModalPlayer {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test("renders for default Props", () => {
      const { getByTestId } = render(
          <VideoModalPlayer {...props}
          />
      );
      expect(getByTestId('test-videplayer-card-wrapper')).toBeDefined();
      expect(VideoModalPlayer.defaultProps.id.videoId).toBeDefined()
  });
  //open_issues returns correctly
  test('open_issues returns correctly' , () => {
      const result = VideoModalPlayer.defaultProps.id.videoId;
      expect(result).toBe('')
  })
});
