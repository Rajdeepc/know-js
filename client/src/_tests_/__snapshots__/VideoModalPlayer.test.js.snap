// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoModalPlayer component> renders VideoModalPlayer Component 1`] = `
<div
  class="video-modal-player"
  data-testid="test-videplayer-card-wrapper"
>
  <iframe
    height="400"
    src="https://www.youtube.com/embed/"
    title=""
    width="100%"
  />
  <div
    class="video-info"
  >
    <div
      class="video-title"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-12"
        >
          <h4 />
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <p>
            
             
          </p>
        </div>
      </div>
      <div
        class="channel row"
      >
        <div
          class="col"
        >
          <div
            class="channel"
          >
            Channel: 
            <a
              href="https://www.youtube.com/channel/"
              rel="noopener noreferrer"
              target="_blank"
            />
          </div>
        </div>
        <div
          class="text-right col"
        >
          Published At: 
          Invalid date
        </div>
      </div>
    </div>
  </div>
</div>
`;
